Description: Build shared lib with bioconda's htslib
 The upstream repo includes htslib as a submodule that gets
 built with the tabixpp code. This patch changes this to use
 bioconda's shared libhts from the archive.
 Also builds a shared library instead of a static one.
Author: Sascha Steinbiss <sascha@steinbiss.name> modified for bioconda by Jon Puritz jpuritz@gmail.com
diff --git a/Makefile b/Makefile
index f2ba44e..7718d76 100644
--- a/Makefile
+++ b/Makefile
@@ -7,30 +7,40 @@
 # this is aimed at making it easy to create packages (Debian packages,
 # FreeBSD ports, MacPorts, pkgsrc, etc.)
 
-CC ?=		cc
-CXX ?= 		c++
-CXXFLAGS ?=	-g -Wall -O2 #-m64 #-arch ppc
+CC ?=		$(CC)
+CXX ?=		$(CXX)
+CXXFLAGS ?=	-g -Wall -O3 #-m64 #-arch ppc
 CXXFLAGS +=	-fPIC
-INCLUDES ?=	-Ihtslib
+INCLUDES ?=	-Ihtslib -I$(PREFIX)/include
 HTS_HEADERS ?=	htslib/htslib/bgzf.h htslib/htslib/tbx.h
 HTS_LIB ?=	htslib/libhts.a
-LIBPATH ?=	-L. -Lhtslib
+LIBPATH ?=	-L. -Lhtslib -L$(PREFIX)/lib
 
-DESTDIR ?=	stage
-PREFIX ?=	/usr/local
+PREFIX ?=	$(prefix)
 STRIP ?=	strip
 INSTALL ?=	install -c
+LN_S ?=		ln -s
 MKDIR ?=	mkdir -p
 AR ?=		ar
 
 DFLAGS =	-D_FILE_OFFSET_BITS=64 -D_USE_KNETFILE
 BIN =		tabix++
-LIB =		libtabix.a
-SOVERSION =	1
-SLIB =		libtabix.so.$(SOVERSION)
+LIB =		libtabixpp.a
+SOVERSION =	0
+SLIB =		libtabixpp.so.$(SOVERSION)
 OBJS =		tabix.o
 SUBDIRS =	.
 
+COMPILER_VERSION := $(shell $(CXX) --version)
+ifneq '' '$(findstring clang,$(COMPILER_VERSION))'
+  SONAME_ARG ?= "-install_name"
+else ifneq '' '$(findstring g++,$(COMPILER_VERSION))'
+  SONAME_ARG ?= "-soname"
+else
+  $(warning Unknown compiler)
+  SONAME_ARG ?= "-soname"
+endif
+
 .SUFFIXES:.c .o
 
 .c.o:
@@ -47,21 +57,19 @@ all-recur lib-recur clean-recur cleanlocal-recur install-recur:
 		cd $$wdir; \
 	done;
 
-all:	$(BIN) $(LIB) $(SLIB)
+all: $(BIN) $(LIB) $(SLIB)
 
-tabix.o: $(HTS_HEADERS) tabix.cpp tabix.hpp
+tabix.o: tabix.cpp tabix.hpp
 	$(CXX) $(CXXFLAGS) -c tabix.cpp $(INCLUDES)
 
-htslib/libhts.a:
-	cd htslib && $(MAKE) lib-static
-
 $(LIB): $(OBJS)
-	$(AR) rs $(LIB) $(OBJS)
+	$(AR) rcs $(LIB) $(OBJS)
 
 $(SLIB): $(OBJS)
-	$(CXX) -shared -Wl,-soname,$(SLIB) -o $(SLIB) $(OBJS)
+	$(CXX) $(CXXFLAGS) -shared -Wl,$(SONAME_ARG),$(SLIB) -o $(SLIB) $(OBJS) \
+		$(INCLUDES) $(LIBPATH) -lhts -lpthread -lm -lz -lcurl -llzma -lbz2
 
-tabix++: $(OBJS) main.cpp $(HTS_LIB)
+tabix++: $(OBJS) main.cpp
 	$(CXX) $(CXXFLAGS) -o $@ main.cpp $(OBJS) $(INCLUDES) $(LIBPATH) \
 		-lhts -lpthread -lm -lz -lcurl -llzma -lbz2
 
@@ -82,7 +90,7 @@ install-strip: install
 cleanlocal:
 	rm -rf $(BIN) $(LIB) $(SLIB) $(OBJS) $(DESTDIR)
 	rm -fr gmon.out *.o a.out *.dSYM $(BIN) *~ *.a tabix.aux tabix.log \
-		tabix.pdf *.class libtabix.*.dylib
+		tabix.pdf *.class libtabixpp.so.* libtabixpp.*.dylib
 	cd htslib && $(MAKE) clean
 
-clean:	cleanlocal-recur
+clean: cleanlocal
